### PR TITLE
Massive docs cleanup and structure clarification

### PR DESCRIPTION
**Pull request by SUPERCILEX**
This PR overhauls our docs structure so that it's clearer and and more easily linkable. It also performs some minor cleanup regarding whitespace and a few other things.

The general idea is to reduce our "Walls of text" by breaking them up with tons of headers.

I've decided to include headers in ToC up to the third (`###`) level, @samtstern do you think that's reasonable?

I'm following the standards [established by GitHub](https://guides.github.com/features/mastering-markdown/#syntax):
- No whitespace before lists
- Titles are lowercase except for the first letter and the h1
- Bullets use `*`

### CODE REVIEW

**Comment by SUPERCILEX** [comment]
I've also moved some files to the `.github` folder, added the standard GitHub code of conduct file with your public email in it, and added us as codeowners. What do you think? We look super official now! 😄👮

File: auth/README.md [hunk 1]
```diff
@@ -246,6 +272,8 @@ startActivityForResult(
     RC_SIGN_IN);
 ```
 
+###### Smart Lock hints
```
Comment (piotrek1543) [inline auth/README.md]: What is a proper name? `Smart Lock` or `SmartLock` as you have below?

File: docs/upgrade-to-2.0.md [hunk 2]
```diff
@@ -12,28 +12,28 @@ For a full, machine-generated compatibility report see the `docs/compat_reports`
     or Twitter SDKs. If you plan to enable either of these providers you should include the
     appropriate SDK in your app to avoid a crash. This saves space for developers only using
     Google/Email/Phone authentication.
-  * **Sign Out and Delete** - The `AuthUI#delete(Activity)` and `AuthUI#signOut(Activity)` 
+  * **Sign Out and Delete** - The `AuthUI#delete(Activity)` and `AuthUI#signOut(Activity)`
```
Comment (piotrek1543) [inline docs/upgrade-to-2.0.md]: I think there should be `out` as you set lowercase to `libraries` before

File: auth/README.md [hunk 3]
```diff
@@ -411,7 +441,7 @@ if (v.getId() == R.id.sign_out) {
 With the integrations provided by FirebaseUI Auth, deleting a user is a multi-stage process:
 
 1. The user must be deleted from Firebase Auth.
-2. Smart Lock for Passwords must be told to delete any existing Credentials for the user, so
+1. Smart Lock for Passwords must be told to delete any existing Credentials for the user, so
```
Comment (piotrek1543) [inline auth/README.md]: Shouldn't be there `2.` as you have `1.` above?

File: auth/README.md [hunk 4]
```diff
@@ -246,6 +272,8 @@ startActivityForResult(
     RC_SIGN_IN);
 ```
 
+###### Smart Lock hints
+
 If you'd like to keep SmartLock's "hints" but disable the saving/retrieving of credentials, then
```
Comment (piotrek1543) [inline auth/README.md]: what about changing SmartLock to `SmartLock`

File: auth/README.md [hunk 5]
```diff
@@ -258,6 +286,8 @@ startActivityForResult(
     RC_SIGN_IN);
 ```
 
+##### Phone number auth customization
```
Comment (piotrek1543) [inline auth/README.md]: What about `authentication` instead of this shortcut?

File: auth/README.md (hunk 1 above)
Comment (SUPERCILEX) [reply auth/README.md]: Thanks! Google calls it `Smart Lock`, will fix.

File: docs/upgrade-to-2.0.md (hunk 2 above)
Comment (SUPERCILEX) [reply docs/upgrade-to-2.0.md]: Yeah, I actually haven't really cleaned up the upgrade guides, will do. (That was an automated find/replace)

File: auth/README.md (hunk 5 above)
Comment (SUPERCILEX) [reply auth/README.md]: SGTM

**Comment by samtstern** [comment]
@SUPERCILEX this all LGTM but if you want we can merge this into `master` instead since it only contains docs fixes and does not depend on the `3.1.3` release.  What do you think?

**Comment by SUPERCILEX** [comment]
@samtstern Habit? Yup, done.

**Comment by samtstern** [comment]
@SUPERCILEX thanks!

**Pull request merged**